### DISCUSSION TITLE
ArtifactHub failed fetching tags from Docker Hub OCI Registry

### DISCUSSION THREAD

**Discussion by sirn** [artifacthub/hub#4064] [General]
Since last week, I've been getting this this error from ArtifactHub for [varnish/varnish-cache](https://artifacthub.io/packages/helm/varnish/varnish-cache) (for [docker.io/varnish/varnish-cache](https://hub.docker.com/r/varnish/varnish-cache)):

```
error getting repository remote digest: GET https://registry-1.docker.io/v2/varnish/varnish-cache/tags/list?n=1000: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:varnish/varnish-cache Type:repository]]
```

After a while, this has changed into:

```
error getting repository remote digest: GET https://auth.docker.io/token?scope=repository%3Avarnish%2Fvarnish-cache%3Apull&service=registry.docker.io: unexpected status code 401 Unauthorized: {"details":"incorrect username or password"}
```

I don't think there's any where we can configure username/password for Docker Hub through ArtifactHub, and the repository is public as far as I can tell. Is there anything else I should check to resolve this issue? 

Thanks!

**Comment by tegioz** [comment]
Hi @sirn 👋

They seem to be punctual errors when pulling data from the Docker Hub.

According to our logs (see table below), it looks like you got 4 of them during the previous 4 weeks. The [Docker system status page](https://www.dockerstatus.com/pages/history/533c6539221ae15e3f000031) reports that they had an incident on Sep 27th, which matches 3 of the errors you got, and some scheduled maintenance on the 24/25th, which matches the fourth. So I'd say nothing to worry about 🙂

Please note that Artifact Hub fetches content from the source repository regularly, so any punctual error like this may trigger an alarm every now and then.

| timestamp | repo | error |
| --- | --- | --- |
| 2024-09-27 21:03:36.432 | varnish | error getting repository remote digest: GET https://auth.docker.io/token?scope=repository%3Avarnish%2Fvarnish-cache%3Apull&service=registry.docker.io: unexpected status code 401 Unauthorized: {"details":"incorrect username or password"} |
| 2024-09-27 20:32:25.791 | varnish | error getting repository remote digest: GET https://auth.docker.io/token?scope=repository%3Avarnish%2Fvarnish-cache%3Apull&service=registry.docker.io: unexpected status code 401 Unauthorized: {"details":"incorrect username or password"} |
| 2024-09-27 03:03:36.419 | varnish | error getting repository remote digest: GET https://auth.docker.io/token?scope=repository%3Avarnish%2Fvarnish-cache%3Apull&service=registry.docker.io: unexpected status code 401 Unauthorized: {"details":"incorrect username or password"} |
| 2024-09-25 00:32:30.953 | varnish | error getting repository remote digest: GET https://registry-1.docker.io/v2/varnish/varnish-cache/tags/list?n=1000: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:varnish/varnish-cache Type:repository]] |

**Reply by sirn** [reply]
Thanks, it looks like this may have been a case of getting pretty unlucky with the pull timing. Is my understanding correct that even after the error goes away the indicator on the Control Panel won't change until the next update?

<img width="508" alt="image" src="https://github.com/user-attachments/assets/49183439-028b-4ab7-8598-2d7755c3acd7">

**Reply by tegioz** [reply]
No worries!

We check repositories every 30 mins, so if something goes wrong for longer than that we'll probably catch it.

About the error in the control panel: yes, that's right 👍

**Reply by sirn** [reply]
OK thank you!
